component/discovery/transformer/types.go

```diff
@@ -0,0 +1,107 @@
+package transformer
+
+import (
+	"fmt"
+
+	"github.com/grafana/regexp"
+)
+
+// Action is the relabelling action to be performed.
+type Action string
+
+// Regexp encapsulates the Regexp type from Grafana's
+// fork of the Go stdlib regexp package.
+// TODO (@tpaschalis) This encapsulation already exists in Prometheus' relabel.Regexp
+// so not sure whether to also move it here for now.
+type Regexp struct {
+	re       *regexp.Regexp
+	original string
+}
+
+// All possible Action values.
+const (
+	Replace   Action = "replace"
+	Keep      Action = "keep"
+	Drop      Action = "drop"
+	HashMod   Action = "hashmod"
+	LabelMap  Action = "labelmap"
+	LabelDrop Action = "labeldrop"
+	LabelKeep Action = "labelkeep"
+	Lowercase Action = "lowercase"
+	Uppercase Action = "uppercase"
+)
+
+var actions = map[Action]struct{}{
+	Replace:   {},
+	Keep:      {},
+	Drop:      {},
+	HashMod:   {},
+	LabelMap:  {},
+	LabelDrop: {},
+	LabelKeep: {},
+	Lowercase: {},
+	Uppercase: {},
+}
+
+// String returns the string representation of the Action type.
+func (a Action) String() string {
+	if _, exists := actions[a]; exists {
+		return string(a)
+	}
+	return "Action:" + string(a)
+}
+
+// MarshalText implements encoding.TextMarshaler for Action.
+func (a Action) MarshalText() (text []byte, err error) {
+	return []byte(a.String()), nil
+}
+
+// UnmarshalText implements encoding.TextUnmarshaler for Action.
+func (a *Action) UnmarshalText(text []byte) error {
+	if _, exists := actions[Action(text)]; exists {
+		*a = Action(text)
+		return nil
+	}
+	return fmt.Errorf("unrecognized action type %s", string(text))
```

Comment:
nit: use `%q` for strings so accidental whitespace is more obvious:

```
unrecognized action type  replace
```

vs 

```
unrecognized action type " replace"
```

Reply:
Nice detail, gonna use it more often!